### ISSUE TITLE
Out of Tree builds: project root discovery

### ISSUE THREAD

**Issue by Thaodan** [MaskRay/ccls#903]
### Observed behavior

When a project uses out of tree builds like in a  _./build/._ directory inside the project folder compile_commands.json are not in the project folder but in that folder. How should such use cases be handled?

### Expected behavior

ccls would describe what is the project root, the build folder or the actual project root or handle projects with out of tree build folders.

### Steps to reproduce

Generate a project that uses an out of tree build. Like for cmake with Cmake:

1. cmake -S $PWD -B build -DCMAKE_EXPORT_COMPILE_COMMANDS=YES
2. How to tell ccsl about the out of tree build?

**Comment by FederAndInk** [comment]
something like this https://github.com/MaskRay/ccls/wiki/Visual-Studio-Code#cclsmisccompilationdatabasedirectory ?

**Comment by FederAndInk** [comment]
see https://github.com/MaskRay/ccls/wiki/Project-Setup#compile_commandsjson you should be able to use `compilationDatabaseDirectory`

**Comment by Thaodan** [comment]
> something like this
> https://github.com/MaskRay/ccls/wiki/Visual-Studio-Code#cclsmisccompilation
> databasedirectory ?
> 
Yes exactly. Can this be more generalized or has this to be supported by each 
editor plugin?
